pyxem/generators/indexation_generator.py

```diff
@@ -278,3 +472,56 @@ def index_vectors(self,
         vectors.hkls = rhkls
 
         return indexation_results
+
+    def refine_best_orientation(self, 
+                                indexation_results, 
+                                rank=0,
+                                index_error_tol=0.2,
+                                method="leastsq"):
+        """Refines the best orientation and assigns hkl indices to diffraction vectors.
+
+        Parameters
+        ----------
+        rank : int
+            The rank of the solution, i.e. rank=2 returns the third best solution
+        index_error_tol : float
+            Max allowed error in peak indexation for classifying it as indexed,
+            calculated as :math:`|hkl_calculated - round(hkl_calculated)|`.
+        method : str
+            Minimization algorithm to use, choose from: 
+            'leastsq', 'nelder', 'powell', 'cobyla', 'least-squares'.
+            See `lmfit` documentation (https://lmfit.github.io/lmfit-py/fitting.html)
+            for more information.
+
+        Returns
+        -------
+        indexation_results : VectorMatchingResults
+            Navigation axes of the diffraction vectors signal containing vector
+            indexation results for each probe position.
+        """
+        vectors = self.vectors
+        library = self.library
+
+        result = indexation_results.map(_refine_best_orientation,
+                                        vectors_cartesian=vectors.cartesian,
+                                        library=library,
+                                        index_error_tol=index_error_tol,
+                                        method=method,
+                                        parallel=False, inplace=False)
+
+        indexation = np.array(result.isig[0].data.tolist(), dtype='object')
+        rhkls = result.isig[1].data
+
+        indexation_results = VectorMatchingResults(indexation)
+        indexation_results.vectors = vectors
+        indexation_results.hkls = rhkls
+        indexation_results = transfer_navigation_axes(indexation_results,
+                                                      vectors.cartesian)
+
+        vectors.hkls = rhkls
+
+        return indexation_results
+
+    def refine_all_orientations(self):
+        pass
```

Comment:
Are you going to do this one before merging?

Reply:
Yes!